rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -142,12 +142,19 @@ async fn try_op(ctx: &mut InterceptorContext, cfg: &mut ConfigBag, interceptors:
         }
     }
 
+    // Save a request checkpoint before we make the request. This will allow us to "rewind"
+    // the request in the case of retry attempts.
+    ctx.save_checkpoint();
     // If you need to retry something more than 255 times then the orchestrator isn't for you.
     for i in 0u8.. {
         // Break from the loop if we can't rewind the request's state. This will always succeed the
         // first time, but will fail on subsequent iterations if the request body wasn't retryable.
-        if !ctx.rewind(cfg) {
-            break;
+        match ctx.rewind(cfg) {
+            RewindResult::Impossible => {
+                break;
+            }
+            r @ RewindResult::Occurred => debug!("{r}"),
+            r @ RewindResult::Unnecessary => debug!("{r}"),
```

Comment:
Are these debug lines useful?

Reply:
I can delete them or make them `trace`s if you like. I think they're useful only insofar as they expose an important part of the request/response lifecycle managed by the orchestrator. However, since the user would never be calling `rewind`, it's not very helpful to them.